maze/test_maze.py

```diff
@@ -282,16 +290,18 @@ def test_maze_heuristic_response(self):
         maze = Maze(n_rows, n_cols, start, end, walls)
         bqm = maze.get_bqm()
 
-        # Sample and test that a response is given
-        sampler = SimulatedAnnealingSampler()
-        response = sampler.sample(bqm, num_reads=1000)
-        response_sample = next(response.samples())
-        self.assertGreaterEqual(len(response), 1)
-
-        # Test heuristic response
-        expected_solution = {'0,1w': 1, '1,1n': 1, '1,1w': 1, '2,0n': 1, '2,1w': 1, '2,2w': 1}
-        fill_with_zeros(expected_solution, n_rows, n_cols, [start, end])
-        self.compare(response_sample, expected_solution)
+        with mock.patch('dwave.system.samplers.DWaveSampler', MockDWaveSampler):
```

Comment:
You could simplify mocking by patching the whole testcase, or at least the complete test. E.g.:
```
@patch(...)
def test(...):
    ...
```
or:
```
@patch(...)
class Test(TestCase):
    ...
```